heartbeat/mariadb-semi-sync

```diff
@@ -0,0 +1,1064 @@
+            master_host=$(parse_slave_info Master_Host $tmpfile)
+            master_user=$(parse_slave_info Master_User $tmpfile)
+            master_port=$(parse_slave_info Master_Port $tmpfile)
+            master_using_gtid=$(parse_slave_info Using_Gtid $tmpfile)
+            master_log_file=$(parse_slave_info Master_Log_File $tmpfile)
+            slave_sql=$(parse_slave_info Slave_SQL_Running $tmpfile)
+            slave_io=$(parse_slave_info Slave_IO_Running $tmpfile)
+            last_errno=$(parse_slave_info Last_Errno $tmpfile)
+            last_error=$(parse_slave_info Last_Error $tmpfile)
+            secs_behind=$(parse_slave_info Seconds_Behind_Master $tmpfile)
+            last_io_errno=$(parse_slave_info Last_IO_Errno $tmpfile)
+            last_io_error=$(parse_slave_info Last_IO_Error $tmpfile)
+            ocf_log debug "MariaDB instance running as a replication slave"
+        else
+            # Instance produced an empty "SHOW SLAVE STATUS" output --
+            # instance is not a slave
+            return $OCF_ERR_GENERIC
+        fi
+
+        return $OCF_SUCCESS
+    fi
+}
+
+check_slave() {
+    # Checks slave status
+    local rc new_master
+
+    get_slave_info
+    rc=$?
+
+    if [ $rc -eq 0 ]; then
+
+        # Check normal errors
+        if [ $last_errno -ne 0 ]; then
+            ocf_exit_reason "MariaDB slave replication has failed ($last_errno): $last_error"
+
+            exit $OCF_ERR_GENERIC
+        fi
+
+        # Check IO Errors, ignore 2003 which indicates a connection failure to the master
+        if [ $last_io_errno -ne 0 ] && [ $last_io_errno -ne 2003 ]; then
+            ocf_exit_reason "MariaDB slave io has failed ($last_io_errno): $last_io_error"
+
+            exit $OCF_ERR_GENERIC
+        fi
+
+        if [ $last_io_errno -eq 2003 ]; then
+            ocf_log warn "MariaDB master not reachable from slave"
+        fi
+        
+        if [ "$slave_io" != 'Yes' ]; then
+            # Not necessarily a bad thing. The master may have
+            # temporarily shut down, and the slave may just be
+            # reconnecting. A warning can't hurt, though.
+            ocf_log warn "MariaDB Slave IO threads currently not running."
+
+            # Sanity check, are we at least on the right master
+            new_master=$($CRM_ATTR_REPL_INFO --query  -q)
+
+            if [ "$master_host" != "$new_master" ]; then
+               # Not pointing to the right master, not good, removing the VIPs
+               set_reader_attr 0
+
+               exit $OCF_SUCCESS
+            fi
+
+        fi
+
+        if [ "$slave_sql" != 'Yes' ]; then
+            # We don't have a replication SQL thread running. Not a
+            # good thing. Try to recoved by restarting the SQL thread
+            # and remove reader vip.  Prevent MariaDB restart.
+            ocf_exit_reason "MariaDB Slave SQL threads currently not running."
+            ocf_log err "See $tmpfile for details"
```

Comment:
I think it's less than ideal to leave $tmpfile lying around, and to assume it'll still be there when someone gets around to reading the logs. Better to log the contents of $tmpfile then remove it.